### PR TITLE
applications: nrf_desktop: Initial project documentation

### PR DESCRIPTION
**Pull request by pdunaj**
Jira:DESK-650

Signed-off-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>

### CODE REVIEW

**Comment by pdunaj** [comment]
This is documentation skeleton.

Hi @greg-fer , can you have a look.
Hi @MarekPieta , as discussed please work on documentation for HW interfaces first. This maybe put on this PR or we can merge everything together once I am back :)

File: applications/nrf_desktop/README.rst [hunk 1]
```diff
@@ -25,24 +35,78 @@ Supported boards:
...
 ********************
+
 .. |sample path| replace:: :file:`applications/nrf_desktop`
 
+The nRF Desktop application is build in the same way to any other NCS
+application or sample.
 .. include:: /includes/build_and_run.txt
 
+There can be several configuration files dedicated for each board. They
+differ by set of selected debugging features. The application support
```
Comment (MarekPieta) [inline applications/nrf_desktop/README.rst]: `debugging features` -> `features`

File: applications/nrf_desktop/README.rst [hunk 2]
```diff
@@ -25,24 +35,78 @@ Supported boards:
 * PCA20045 - nRF Desktop casual mouse reference design (nrf52810)
 * PCA10059 - nRF Desktop dongle reference design
 
+The application was designed to allow easy porting to a new HW. Please check
+DUPA to get information about the device porting steps.
```
Comment (MarekPieta) [inline applications/nrf_desktop/README.rst]: Add ref to porting guide

File: applications/nrf_desktop/README.rst [hunk 3]
```diff
@@ -25,24 +35,78 @@ Supported boards:
...
+
+
+Dependencies
+************
+
+This application uses the following |NCS| libraries and drivers:
+
+    * :ref:`event_manager`
```
Comment (MarekPieta) [inline applications/nrf_desktop/README.rst]: mention profiler as debugging option ?

File: applications/nrf_desktop/README.rst [hunk 4]
```diff
@@ -8,15 +8,25 @@ Nordic Semiconductor HID reference design.
 Overview
 ********
 
-This application is a reference design of a HID device.
-
 The code is currently work-in-progress and as such is not fully functional, verified, or
```
Comment (greg-fer) [inline applications/nrf_desktop/README.rst]: I'd move this sentence to just before  Requirements, and highlight it with a note.

File: applications/nrf_desktop/README.rst [hunk 5]
```diff
@@ -8,15 +8,25 @@ Nordic Semiconductor HID reference design.
...
 supported for product development.
 
+Welcome to the firmware documentation for nRF52 Desktop, a reference design of
```
Comment (greg-fer) [inline applications/nrf_desktop/README.rst]: No need to welcome, I'd just state what nRF52 Desktop is.

File: applications/nrf_desktop/README.rst [hunk 6]
```diff
@@ -8,15 +8,25 @@ Nordic Semiconductor HID reference design.
...
+a HID device.
+
+Depending on a configuration this application can function as mouse,
+gaming mouse or keyboard connected to host via BLE and/or USB.
+
+Project supports common input HW interfaces i.e. motion sensors, rotation
```
Comment (greg-fer) [inline applications/nrf_desktop/README.rst]: "i.e." meaning "that is"? If so, this is a finished list, so let's transform it into a bullet point list.

File: applications/nrf_desktop/README.rst [hunk 7]
```diff
@@ -8,15 +8,25 @@ Nordic Semiconductor HID reference design.
...
+sensors and buttons scanning module.
+Firmware can be configured at run-time using dedicated configuration channel
+realized with HID feature report. The same channel is used to pass DFU packets.
+
+
 Requirements
 ************
 
-Supported boards:
+The project contains configuration files for following boards:
```
Comment (greg-fer) [inline applications/nrf_desktop/README.rst]: Requirements and supported boards focus on two different things: not all boards mentioned in the list are required to use the firmware. Can we rephrase `Requirements` to `Supported boards`?

Comment (greg-fer) [inline applications/nrf_desktop/README.rst]: Also, how do I know which file is for which board? Where can I find these files?

File: applications/nrf_desktop/README.rst [hunk 8]
```diff
@@ -25,24 +35,78 @@ Supported boards:
...
 Building and Running
 ********************
+
 .. |sample path| replace:: :file:`applications/nrf_desktop`
 
+The nRF Desktop application is build in the same way to any other NCS
+application or sample.
 .. include:: /includes/build_and_run.txt
 
+There can be several configuration files dedicated for each board. They
```
Comment (greg-fer) [inline applications/nrf_desktop/README.rst]: Would this sentence and the following one sound more appropriate in the section about supported boards?

File: applications/nrf_desktop/README.rst (hunk 4 above)
Comment (pdunaj) [reply applications/nrf_desktop/README.rst]: done

File: applications/nrf_desktop/README.rst (hunk 5 above)
Comment (pdunaj) [reply applications/nrf_desktop/README.rst]: done

File: applications/nrf_desktop/README.rst (hunk 6 above)
Comment (pdunaj) [reply applications/nrf_desktop/README.rst]: it should actually be e.g. or better "like"

File: applications/nrf_desktop/README.rst (hunk 7 above)
Comment (pdunaj) [reply applications/nrf_desktop/README.rst]: I don't mind. I though that this is the Zephyr's convention. Samples and assert tracker mention boards in requirements.

Comment (pdunaj) [reply applications/nrf_desktop/README.rst]: I plan to give more detail in configuration guide and would prefer to limit the number of information given in this section to supported boards.

File: applications/nrf_desktop/README.rst (hunk 3 above)
Comment (pdunaj) [reply applications/nrf_desktop/README.rst]: ok

File: applications/nrf_desktop/README.rst (hunk 8 above)
Comment (pdunaj) [reply applications/nrf_desktop/README.rst]: The context is different - we need to tell which of various possible variants is chosen for building. When board is added it should contain at least one configuration  (for one of the build variants).

Comment (pdunaj) [reply applications/nrf_desktop/README.rst]: I will rephrase it so maybe meaning will be easier to understand.

File: applications/nrf_desktop/README.rst (hunk 1 above)
Comment (pdunaj) [reply applications/nrf_desktop/README.rst]: ok

File: applications/nrf_desktop/README.rst (hunk 7 above)
Comment (greg-fer) [reply applications/nrf_desktop/README.rst]: @b-gent , can you confirm this is a Zephyr convention?

File: applications/nrf_desktop/README.rst [hunk 9]
```diff
@@ -8,45 +8,108 @@ Nordic Semiconductor HID reference design.
...
+The project contains configuration files for following boards:
+
+    * PCA10056 (nRF52840 DK) - a sample where the mouse is emulated using DK buttons
+    * PCA20041 - nRF Desktop gaming mouse reference design
+    * PCA20037 - nRF Desktop keyboard reference design
+    * PCA20044 - nRF Desktop casual mouse reference design (nRF52832)
+    * PCA20045 - nRF Desktop casual mouse reference design (nRF52810)
+    * PCA10059 - nRF Desktop dongle reference design
+
+The application was designed to allow easy porting to a new HW. Please check
```
Comment (greg-fer) [inline applications/nrf_desktop/README.rst]:
```suggestion
The application was designed to allow easy porting to a new HW.
```

File: applications/nrf_desktop/README.rst [hunk 10]
```diff
@@ -8,45 +8,108 @@ Nordic Semiconductor HID reference design.
...
+:ref:`porting_guide` to get information about the device porting steps.
```
Comment (greg-fer) [inline applications/nrf_desktop/README.rst]:
```suggestion
Check :ref:`porting_guide` for information about the device porting steps.
```

File: applications/nrf_desktop/README.rst [hunk 11]
```diff
@@ -8,45 +8,108 @@ Nordic Semiconductor HID reference design.
...
 Building and Running
 ********************
+
 .. |sample path| replace:: :file:`applications/nrf_desktop`
 
+The nRF Desktop application is build in the same way to any other NCS
+application or sample.
 .. include:: /includes/build_and_run.txt
 
+Configuration options set depends on selected build type. Project supports
```
Comment (greg-fer) [inline applications/nrf_desktop/README.rst]:
```suggestion
Configuration option set depends on the selected build type.
```

File: applications/nrf_desktop/README.rst [hunk 12]
```diff
@@ -8,45 +8,108 @@ Nordic Semiconductor HID reference design.
...
+following types.
```
Comment (greg-fer) [inline applications/nrf_desktop/README.rst]:
```suggestion
The project supports the following types:
```

File: applications/nrf_desktop/README.rst [hunk 13]
```diff
@@ -8,45 +8,108 @@ Nordic Semiconductor HID reference design.
...
+
+    * ZRelease - release version of the application with no debugging features
+    * ZDebug - the same functionality as ZRelease but with debug options enabled
+    * ZDebugWithShell - the same set as ZDebug plus shell is enabled
+    * ZReleaseMCUBoot - the same set as ZRelease but support of MCUBoot is enabled
+    * ZDebugMCUBoot - the same set as ZDebug but support of MCUBoot is enabled
+
+Build type is specified using the cmake variable CMAKE_BUILD_TYPE:
```
Comment (greg-fer) [inline applications/nrf_desktop/README.rst]: Looks like a codeblock is missing? If not, use period at the end.

File: applications/nrf_desktop/README.rst [hunk 14]
```diff
@@ -8,45 +8,108 @@ Nordic Semiconductor HID reference design.
...
+
+
+Application configuration
+*************************
+
+For detailed information about the application configuration and how to port
+it to a different HW platform please refer to the :ref:`porting_guide`.
```
Comment (greg-fer) [inline applications/nrf_desktop/README.rst]:
```suggestion
it to a different hardware platform, see :ref:`porting_guide`.
```

File: applications/nrf_desktop/README.rst [hunk 15]
```diff
@@ -8,45 +8,108 @@ Nordic Semiconductor HID reference design.
...
+
 
 Firmware architecture
 *********************
 
-Explanation of how the application code is designed and how it works.
+The nRF Desktop application design aimed at configurability and extensibility
```
Comment (greg-fer) [inline applications/nrf_desktop/README.rst]:
```suggestion
The nRF Desktop application design aims at configurability and extensibility
```

File: applications/nrf_desktop/README.rst [hunk 16]
```diff
@@ -8,45 +8,108 @@ Nordic Semiconductor HID reference design.
...
+while allowing it to be responsive.
```
Comment (greg-fer) [inline applications/nrf_desktop/README.rst]:
```suggestion
while being responsive.
```

File: applications/nrf_desktop/README.rst [hunk 17]
```diff
@@ -8,45 +8,108 @@ Nordic Semiconductor HID reference design.
...
+
+The application architecture is modular and event driven. Modularity means that
```
Comment (greg-fer) [inline applications/nrf_desktop/README.rst]:
```suggestion
The application architecture is modular and event-driven.
```

File: applications/nrf_desktop/README.rst [hunk 18]
```diff
@@ -8,45 +8,108 @@ Nordic Semiconductor HID reference design.
...
+parts of application functionality were separated into isolated modules.
```
Comment (greg-fer) [inline applications/nrf_desktop/README.rst]:
```suggestion
This means that parts of application functionality were separated into isolated modules
```

File: applications/nrf_desktop/README.rst [hunk 19]
```diff
@@ -8,45 +8,108 @@ Nordic Semiconductor HID reference design.
...
+Modules communicate with each other using application events.
```
Comment (greg-fer) [inline applications/nrf_desktop/README.rst]:
```suggestion
that communicate with each other using application events.
```

**Comment by pdunaj** [comment]
@b-gent

**Pull request merged**